scripts/g.extension.all/g.extension.all.py

```diff
@@ -77,6 +87,127 @@ def get_extensions():
     return ret
 
 
+def urlopen(url, *args, **kwargs):
+    """Wrapper around urlopen. Same function as 'urlopen', but with the
+    ability to define headers.
+    """
+    request = urlrequest.Request(url, headers=HEADERS)
+    return urlrequest.urlopen(request, *args, **kwargs)
+
+
+def download_modules_xml_file(url, response_format, *args, **kwargs):
+    """Generates JSON file containing the download URLs of the official
+    Addons
+
+    :param str url: url address
+    :param str response_format: content type
+
+    :return response: urllib.request.urlopen response object or None
+    """
+    try:
+        response = urlopen(url, *args, **kwargs)
+
+        if not response.code == 200:
+            index = HTTP_STATUS_CODES.index(response.code)
+            desc = HTTP_STATUS_CODES[index].description
+            gscript.fatal(
+                _(
+                    "Download file from <{url}>, "
+                    "return status code {code}, "
+                    "{desc}".format(
+                        url=url,
+                        code=response.code,
+                        desc=desc,
+                    ),
+                ),
+            )
+        if response_format not in response.getheader("Content-Type"):
+            gscript.fatal(
+                _(
+                    "Wrong downloaded file format. "
+                    "Check url <{url}>. Allowed file format is "
+                    "{response_format}.".format(
+                        url=url,
+                        response_format=response_format,
+                    ),
+                ),
+            )
+        return response
+
+    except HTTPError as err:
+        if err.code == 404:
+            gscript.fatal(
+                _(
+                    "The download of the modules.xml file "
+                    "from the server was not successful. "
+                    "File on the server <{url}> doesn't "
+                    "exists.".format(url=url),
+                ),
+            )
+        else:
+            return download_modules_xml_file(
+                url=url,
+                response_format=response_format,
+            )
+    except URLError:
+        gscript.fatal(
+            _(
+                "Download file from <{url}>, "
+                "failed. Check internet connection.".format(
+                    url=url,
+                ),
+            ),
+        )
+
+
+def find_addon_name(addons):
+    """Find correct addon name if addon is a multi-addon
+    e.g. wx.metadata contains multiple modules g.gui.cswbrowser etc.
+
+    Examples:
+    - for the g.gui.cswbrowser module the wx.metadata addon name is
+    returned
+    - for the i.sentinel.download module the i.sentinel.download addon
```

Comment:
```suggestion
    - for the i.sentinel.download module the i.sentinel addon
```
The parent multi-addon is to be called, right?

Reply:
Yes, you are right.